compute_sdk/globus_compute_sdk/sdk/bash_function.py

```diff
@@ -0,0 +1,245 @@
+import typing as t
+
+
+class BashResult:
+
+    def __init__(
+        self,
+        cmd: str,
+        stdout: str,
+        stderr: str,
+        returncode: int,
+        exception_name: t.Optional[str] = None,
+    ):
+        """
+
+        Parameters
+        ----------
+        cmd: str
+            formatted command line string that was executed on the endpoint
+
+        stdout: str
+            multiline (default 1K) snippet of stdout
+
+        stderr: str
+            multiline (default 1K) snippet of stderr
+
+        returncode: int
+            Return code from command execution
+
+        exception_name
+        """
+        self.cmd = cmd
+        self.stdout = stdout
+        self.stderr = stderr
+        self.returncode = returncode
+        self.exception_name = exception_name
+
+    def __str__(self):
+        return f"Command {self.cmd} returned with exit status: {self.returncode}"
+
+    def __repr__(self):
+        return (
+            f"{self.__class__.__name__}(returncode={self.returncode}, cmd={self.cmd})"
+        )
+
+
+class BashFunction:
+
+    def __init__(
+        self,
+        cmd: str,
+        stdout: t.Optional[str] = None,
+        stderr: t.Optional[str] = None,
+        walltime: t.Optional[float] = None,
+        # This could be a os.Pathlike, but check windows->unix transition
+        rundir: t.Optional[str] = None,
+        run_in_sandbox: bool = True,
```

Comment:
I think we should drop these and rely on the already existing `working_dir` and `run_in_sandbox` endpoint configuration variables.

Reply:
I've removed these options. I've updated docs to recommend setting `run_in_sandbox` on the endpoint config for BashFunctions, and to return a warning message in the stderr output if sandboxing isn't enabled.